docs/testing/unit-test-mock-library.md

```diff
@@ -0,0 +1,342 @@
+# Unit Test Mock Library
+
+***Table of Contents***
+
+  - [Introduction](#introduction)
+  - [General Usage](#general-usage)  
+  - Functions
+    - [buildMockCell](#buildmockcell)
+    - [buildMockNotebook](#buildmocknotebook)
+    - [mockServiceWizardLookup](#mockservicewizardlookup)
+    - [mockJsonRpc1Call](#mockjsonrpc1call)
+    - [mockAuthRequest](#mockauthrequest)
+    - [setAuthToken](#setauthtoken)
+    - [clearAuthToken](#clearauthtoken)
+
+## Introduction
+The Narrative Mocks library is a tool for mocking various components of the Narrative interface while testing. It's mostly helpful for reducing boilerplate needed for testing Jupyter notebook cell extensions, and mocking KBase service calls.
+
+Below is a list of all currently available functions with examples.
+
+## General Usage
+```Javascript
+define([
+    'narrativeMocks'
+], (Mocks) => {
+    'use strict';
+    // all the mock functions are now available under the Mocks object.
+});
+```
+
+## buildMockCell
+`buildMockCell(cellType, kbaseCellType, data)`  
+This function builds a mock Jupyter notebook cell given a Jupyter notebook cell type and an optional KBase cell type. It includes the full structure needed to portray a cell, including DOM elements.
```

Comment:
In case you want picky grammatical comments: add a comma after "a mock Jupyter notebook cell"

Should probably say that the DOM structure is simplified as it doesn't recapitulate the full cell structure.

Reply:
Yes, be picky! I wasn't really as diligent as I could have about going back and revising this.